go/runtime/client/watcher.go

```diff
@@ -103,64 +121,112 @@ func (w *blockWatcher) watch() {
 	}
 	defer blocksSub.Close()
 
-	// If we were just started, refresh the committee information from any
-	// block, otherwise just from epoch transition blocks.
-	var gotInitialCommittee bool
+	// Start watching consensus blocks.
+	consensusBlocks, consensusBlocksSub, err := w.common.consensus.WatchBlocks(w.common.ctx)
+	if err != nil {
+		w.Logger.Error("failed to subscribe to consensus blocks",
+			"err", err,
+		)
+		return
+	}
+	defer consensusBlocksSub.Close()
+
+	// latestHeight contains the latest known consensus block height.
+	var latestHeight int64
 	// latestGroupVersion contains the latest known committee group version.
 	var latestGroupVersion int64
+	// Wait for first consensus block before proceeding.
+	select {
+	case <-w.stopCh:
+		return
+	case <-w.common.ctx.Done():
+		return
+	case blk := <-consensusBlocks:
+		latestHeight = blk.Height
+		latestGroupVersion, err = w.getGroupVersion(blk.Height)
+		if err != nil {
+			w.Logger.Error("failed querying for latest group version",
+				"err", err,
+			)
+			return
+		}
+	}
+
 	for {
 		// Wait for stuff to happen.
 		select {
 		case blk := <-blocks:
-			// Check block.
-			w.checkBlock(blk.Block)
+			w.toBeChecked = append(w.toBeChecked, blk.Block)
 
-			// If this is the initial block or an epoch transition block,
-			// update latest known group version and resend all transactions.
-			if gotInitialCommittee && blk.Block.Header.HeaderType != block.EpochTransition {
-				continue
+			var failedBlocks []*block.Block
+			for _, b := range w.toBeChecked {
+				if err = w.checkBlock(b); err != nil {
+					w.Logger.Error("error checking block",
+						"err", err,
+						"block", b,
```

Comment:
Should this maybe only log the round?

Reply:
Yeah change to log `round` instead of the `block`.